homeassistant/components/pvpc_hourly_pricing/config_flow.py

```diff
@@ -33,36 +60,168 @@ async def async_step_user(
+            self._use_api_token = user_input[CONF_USE_API_TOKEN]
+            return await self._async_verify("reauth_confirm", data_schema)
+        return self.async_show_form(step_id="reauth_confirm", data_schema=data_schema)
+
+
+class PVPCOptionsFlowHandler(config_entries.OptionsFlowWithConfigEntry):
     """Handle PVPC options."""
 
-    def __init__(self, config_entry: config_entries.ConfigEntry) -> None:
-        """Initialize options flow."""
-        self.config_entry = config_entry
+    _power: float | None = None
+    _power_p3: float | None = None
+
+    async def async_step_api_token(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle optional step to define API token for extra sensors."""
+        if user_input is not None and user_input.get(CONF_API_TOKEN):
+            return self.async_create_entry(
+                title="",
+                data={
+                    ATTR_POWER: self._power,
+                    ATTR_POWER_P3: self._power_p3,
+                    CONF_USE_API_TOKEN: True,
+                    CONF_API_TOKEN: user_input[CONF_API_TOKEN],
+                },
+            )
+
+        # Fill options with entry data
+        api_token = self.options.get(
+            CONF_API_TOKEN, self.config_entry.data.get(CONF_API_TOKEN)
+        )
+        return self.async_show_form(
+            step_id="api_token",
+            data_schema=vol.Schema(
+                {vol.Required(CONF_API_TOKEN, default=api_token): str}
+            ),
+            description_placeholders={"mail_to_link": _MAIL_TO_LINK},
+        )
 
     async def async_step_init(
         self, user_input: dict[str, Any] | None = None
     ) -> FlowResult:
         """Manage the options."""
         if user_input is not None:
+            if user_input[CONF_USE_API_TOKEN]:
+                self._power = user_input[ATTR_POWER]
+                self._power_p3 = user_input[ATTR_POWER_P3]
+                return await self.async_step_api_token(user_input)
             return self.async_create_entry(title="", data=user_input)
 
         # Fill options with entry data
-        power = self.config_entry.options.get(
-            ATTR_POWER, self.config_entry.data[ATTR_POWER]
-        )
-        power_valley = self.config_entry.options.get(
+        power = self.options.get(ATTR_POWER, self.config_entry.data[ATTR_POWER])
+        power_valley = self.options.get(
             ATTR_POWER_P3, self.config_entry.data[ATTR_POWER_P3]
         )
+        use_api_token = self.options.get(
+            CONF_USE_API_TOKEN,
+            self.config_entry.data.get(CONF_USE_API_TOKEN, False),
+        )
         schema = vol.Schema(
             {
                 vol.Required(ATTR_POWER, default=power): VALID_POWER,
                 vol.Required(ATTR_POWER_P3, default=power_valley): VALID_POWER,
+                vol.Required(CONF_USE_API_TOKEN, default=use_api_token): bool,
```

Comment:
I can see why you would use `CONF_USE_API_TOKEN` here by why not let the default be the existence of `CONF_API_TOKEN` rather than storing it in config?

Reply:
> why not let the default be the existence of CONF_API_TOKEN rather than storing it in config?

Please see the changes in the last commit, I've done just that: config-flow does not change, but the boolean flag is not stored in entry.data nor entry.options 👍